src/transformers/models/oneformer/modeling_oneformer.py

```diff
@@ -0,0 +1,3204 @@
+            (0 for the negative class and 1 for the positive class).
+
+    Returns:
+        `torch.Tensor`: The computed loss between each pairs.
+    """
+    inputs = inputs.sigmoid().flatten(1)
+    numerator = 2 * torch.einsum("nc,mc->nm", inputs, labels)
+    # using broadcasting to get a [num_queries, NUM_CLASSES] matrix
+    denominator = inputs.sum(-1)[:, None] + labels.sum(-1)[None, :]
+    loss = 1 - (numerator + 1) / (denominator + 1)
+    return loss
+
+
+# refactored from original implementation
+def pair_wise_sigmoid_ce_loss(inputs: Tensor, labels: Tensor) -> Tensor:
+    r"""
+    A pair wise version of the ce loss, see `sigmoid_ce_loss` for usage.
+
+    Args:
+        inputs (`torch.Tensor`):
+            A tensor representing a mask
+        labels (`torch.Tensor`):
+            A tensor with the same shape as inputs. Stores the binary classification labels for each element in inputs
+            (0 for the negative class and 1 for the positive class).
+
+    Returns:
+        `torch.Tensor`: The computed loss between each pairs.
+    """
+    hw = inputs.shape[1]
+    pos = nn.functional.binary_cross_entropy_with_logits(inputs, torch.ones_like(inputs), reduction="none")
+    neg = nn.functional.binary_cross_entropy_with_logits(inputs, torch.zeros_like(inputs), reduction="none")
+    loss = torch.einsum("nc,mc->nm", pos, labels) + torch.einsum("nc,mc->nm", neg, (1 - labels))
+    return loss / hw
+
+
+def calculate_uncertainty(logits: Tensor) -> Tensor:
+    """
+    We estimate uncerainty as L1 distance between 0.0 and the logit prediction in 'logits' for the
+        foreground class in `classes`.
+
+    Args:
+        logits (Tensor): A tensor of shape (R, 1, ...) for class-specific or
+            class-agnostic, where R is the total number of predicted masks in all images and C is the number of
+            foreground classes. The values are logits.
+    Returns:
+        scores (Tensor): A tensor of shape (R, 1, ...) that contains uncertainty scores with
+            the most uncertain locations having the highest uncertainty score.
+    """
+    if not logits.shape[1] == 1:
+        raise ValueError(f"logits shape at dimension 1 must be 1, found {logits.shape[1]}")
+    gt_class_logits = logits.clone()
+    return -(torch.abs(gt_class_logits))
+
+
+def point_sample(input: Tensor, point_coords: Tensor, **kwargs) -> Tensor:
+    """
+    A wrapper around :function:`torch.nn.functional.grid_sample` to support 3D point_coords tensors. Unlike
+    `nn.functional.grid_sample` it assumes `point_coords` to lie inside [0, 1] x [0, 1] square.
+
+    Args:
+        input (Tensor): A tensor of shape (N, C, H, W) that contains features map on a H x W grid.
+        point_coords (Tensor):
+            A tensor of shape (N, P, 2) or (N, Hgrid, Wgrid, 2) that contains [0, 1] x [0, 1] normalized point
+        coordinates.
```

Comment:
```suggestion
        input (`torch.FloatTensor`):
            A tensor of shape (batch_size, num_channels, height, width) that contains the feature maps on a height x width grid.
        point_coords (Tensor):
            A tensor of shape (N, P, 2) or (N, Hgrid, Wgrid, 2) that contains [0, 1] x [0, 1] normalized point
        coordinates.
```
Would be great to make the one-letter variables more explicit + use our [documentation syntax](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation)